config/action.d/firewallcmd-allports.conf

```diff
@@ -0,0 +1,52 @@
+# Fail2Ban configuration file
+#
+# Author: Donald Yandt 
+# Because of the --remove-rules in stop this action requires firewalld-0.3.8+
+
+
+[INCLUDES]
+
+before = iptables-blocktype.conf
+
+actionstart = firewall-cmd --direct --add-chain ipv4 filter f2b-<name>
+              firewall-cmd --direct --add-rule ipv4 filter f2b-<name> 1000 -j RETURN
+              firewall-cmd --direct --add-rule ipv4 filter <chain> 0 -j f2b-<name>
+
+actionstop = firewall-cmd --direct --remove-rule ipv4 filter <chain> 0 -j f2b-<name>
+             firewall-cmd --direct --remove-rules ipv4 filter f2b-<name>
+             firewall-cmd --direct --remove-chain ipv4 filter f2b-<name>
+
+
+# Note:  uses regular expression word boundaries '\b' 
+# Example actioncheck: firewall-cmd --direct --get-chains ipv4 filter | grep -q '\bf2b-recidive\b'
+
+actioncheck = firewall-cmd --direct --get-chains ipv4 filter | grep -q '\bf2b-<name>\b'
```

Comment:
Okay I wasn't sure exactly when actioncheck was called but if it's only called after adding a chain then '\s' & '$' would be better suited. I was using firewallcmd and grep -c to come up with a regexp. 

Also the reason I tried to clean up a bit was because I'm not using Git and I ended up with 4 patches since I'm not use to the web api...so please bear with me lol.